extensions/cast/src/main/java/com/google/android/exoplayer2/ext/cast/CastPlayer.java

```diff
@@ -145,42 +145,57 @@ public CastPlayer(CastContext castContext) {
     pendingSeekWindowIndex = C.INDEX_UNSET;
     pendingSeekPositionMs = C.TIME_UNSET;
     updateInternalState();
+
+    if (remoteMediaClient != null) {
+      try {
+        remoteMediaClient.addListener(statusListener);
+        remoteMediaClient.addProgressListener(statusListener, PROGRESS_REPORT_PERIOD_MS);
+        updateInternalState();
+      } catch (NullPointerException ignored) {
+      }
+    }
   }
 
   // Media Queue manipulation methods.
 
   /**
    * Loads a single item media queue. If no session is available, does nothing.
    *
-   * @param item The item to load.
+   * @param item       The item to load.
    * @param positionMs The position at which the playback should start in milliseconds relative to
-   *     the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
-   *     starts at position 0.
+   *                   the start of the item at {@code startIndex}. If {@link C#TIME_UNSET} is passed, playback
+   *                   starts at position 0.
    * @return The Cast {@code PendingResult}, or null if no session is available.
    */
   public PendingResult<MediaChannelResult> loadItem(MediaQueueItem item, long positionMs) {
-    return loadItems(new MediaQueueItem[] {item}, 0, positionMs, REPEAT_MODE_OFF);
```

Comment:
Fix for:
3. ArrayIndexOutOfBoundsException: length=2; index=2 when setting new media sources and some other problems with wron current position.